Dark/Light Mode button implementation for issue #7

CoderPrime99 opened the pull request:
# Pull Request Overview

## 📝 Summary
This PR accomplishes the issue presented of making the functionality of ThemeToggle to work, changing from dark mode to light mode in the available routes of this project.

### 🪧 Related Issues
- Closes -- Create dark mode and light mode button #7

### 🏁 Type of Change
Mark with an `x` all the checkboxes that apply (like `[x]`).

- [ ] 📝 Documentation (updates to README, docs, or comments)
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x ] 👌 Enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

🔄 Changes Made

- Header.tsx: Added the ThemeToggle component and removed the user icon (<FaUserCircle />) from the connect/disconnect button.
- ThemeToggle.tsx: Modified theme initialization to retrieve the stored theme from localStorage and update it accordingly. Adjusted the toggleTheme function to use the functional form of setTheme for consistent state updates.
- Global.css: Added background styles for light and dark modes. Light mode has no background image, while dark mode retains the background image.
- Layout.css: Added styles for the text-white and languageIcon classes to ensure they remain visible in both light and dark modes.
- LanguageSwitcher.tsx: languageIcon was added to the beginning of the className so it could be changed in the layout.css file.
- 
🚀 Implementation Details

Header.tsx: 
Added <ThemeToggle /> in place of the user icon to allow switching between light and dark modes.

ThemeToggle.tsx: 
Initialized the theme state from localStorage to persist user preferences. The theme updates are now stored in localStorage upon toggle.
Updated toggleTheme to use the functional form of setTheme for reliable state transitions.

Global.css:
Added background rules for both light and dark modes using CSS variables.

Layout.css:
Ensured that text and icons maintain proper color styling in both light and dark modes.

LanguageSwitcher.tsx: 
Ensured that icon inside the Language Switcher had a class name to be identified in the layout.css so I could directly modify it.


🛠 Technical Notes

Theme Persistence: The theme is now saved in localStorage to persist across page reloads.
CSS Variables: Utilized :root for light mode and .dark class for dark mode to control background styles.
State Management: Used the functional form of setState in toggleTheme to avoid potential issues with asynchronous state updates.

✅ Tests Results

Performed the following tests to verify the functionality of the changes:

- Tested the light and dark mode toggle on all available routes to ensure the background and text colors switched correctly.
- Verified the persistence of the selected theme across page reloads by checking localStorage.
- Confirmed the removal of the user icon and proper placement of the ThemeToggle button.
- Ensured the languageIcon and text-white styles adapted properly to both modes.

Test Coverage

 ✅ Unit Tests (Not applicable for CSS/visual changes)
 📨 Integration Tests (Not applicable for CSS/visual changes)
 📟 Manual Testing

📸 Evidence

![Home Page Light Mode](https://github.com/user-attachments/assets/ca03da21-4bd0-4662-8a24-b2069e43390c)

![Home Page Dark Mode](https://github.com/user-attachments/assets/c428590f-8153-496b-8941-01de82977ffb)

🔍 Testing Notes

- Verified the ThemeToggle button was functional even during rapid clicks to ensure state consistency.
- Checked edge cases, including:
     - Default theme on a fresh visit (light or dark based on localStorage or the absence of prior settings).
     - Correct fallback behavior if localStorage is unavailable.
     - Behavior when the browser does not support CSS variables.
     
🔜 Next Steps

Review feedback and apply changes if needed.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a `ThemeToggle` component allowing users to switch between light and dark themes.
	- Enhanced background styling for light and dark modes in the application.

- **Bug Fixes**
	- Improved theme persistence across sessions by retrieving the current theme from local storage.

- **Style**
	- Updated styling for body backgrounds and added new utility classes for text colors based on the selected theme.
	- Enhanced styling for the language icon in the `LanguageSwitcher` component.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

CoderPrime99 commented on src/components/header/ThemeToggle.tsx:
```diff
@@ -3,19 +3,22 @@ import { LuMoonStar } from 'react-icons/lu';
 
 export default function ThemeToggle() {
-  const [theme, setTheme] = useState('light');
+  const [theme, setTheme] = useState(() => localStorage.getItem('theme') || 'light'); // Retrieves from local storage
```
The file of ThemeToggle was modified with the solution presented as it was a better approach, the page was also tested with this change and it works.

CoderPrime99 commented on src/components/header/ThemeToggle.tsx:
```diff
@@ -3,19 +3,22 @@ import { LuMoonStar } from 'react-icons/lu';
     } else {
       root.classList.remove('dark');
+      localStorage.setItem('theme', 'light'); // Stores the selected theme
     }
```
The file of ThemeToggle was modified with the solution presented as it was a better approach, the page was also tested with this change and it works.

salazarsebas commented:
Please check the color of the letters when in light mode. Otherwise very good work @CoderPrime99

sasasamaes reviewed (approved):
Great Work @CoderPrime99 🚀

CoderPrime99 commented:
Hello @salazarsebas, I have modified the global.css and tailwind.config.ts files, not the layout.css anymore since that file regenerates with the install process, now the text that is white in dark mode, is changing to black correctly when changing to white mode, and the symbol for the language switcher is remaining white in both modes so it is visible based on the design previoulsy made.

So, the files I changed for this to work are: 

- Header.tsx
- LanguageSwitcher.tsx
- ThemeToggle.tsx
- Global.css
- tailwind.config.ts

![image](https://github.com/user-attachments/assets/5abf2729-3df3-452d-a64d-dcfad4bb5828)
![image](https://github.com/user-attachments/assets/28c424d2-593d-46f4-b250-893fcd4f3232)

salazarsebas reviewed (approved):
LGTM @CoderPrime99 🚜🫡
